nncf/experimental/common/pruning/operations.py

```diff
@@ -275,6 +283,34 @@ def _get_output_mask(cls, input_masks: List[Optional[PropagationMask]]) -> Optio
                             group.invalidate()
         return output_mask
 
+    @staticmethod
+    def _propagate_mask_single(
+        input_masks: List[Optional[PropagationMask]], input_shapes: List[Tuple[int, ...]]
+    ) -> Optional[PropagationMask]:
+        """
+        Attempts to propagate a mask in case of one input mask In case one of the masks is None.
+
+        :param input_masks: List of propagation masks for each input of the element-wise operation
+        :param input_shapes: List of tensor shapes for each input.
+        :return: A instance of PropagationMask or None.
+        """
+        none_mask_ind = input_masks.index(None)
+        mask_ind = 0 if none_mask_ind else 1
+
+        dims_diff = len(input_shapes[mask_ind]) - len(input_shapes[none_mask_ind])
+        padded_none_mask_shape = (1,) * dims_diff + input_shapes[none_mask_ind]
+
+        for dim in input_masks[mask_ind].dim_groups_map:
+            if padded_none_mask_shape[dim] != 1:
+                return None
+
+        output_mask = PropagationMask()
+        dims_shift = min(dims_diff, 0)
```

Comment:
Do you cover the case when it's negative and can be propagated?

Reply:
https://github.com/openvinotoolkit/nncf/pull/2302/files#diff-6215ab3079ec3de37965b16b9c0b921a618e911bb6b72ecc4ffb1959c27e1ca7R147